src/Mvc/src/Microsoft.AspNetCore.Mvc.NewtonsoftJson/DependencyInjection/NewtonsoftJsonMvcCoreBuilderExtensions.cs

```diff
@@ -0,0 +1,104 @@
+{
+    public static class NewtonsoftJsonMvcCoreBuilderExtensions
+    {
+        /// <summary>
+        /// Configures Newtonsoft.Json specific features such as input and output formatters.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcCoreBuilder"/>.</param>
+        /// <returns>The <see cref="IMvcCoreBuilder"/>.</returns>
+        public static IMvcCoreBuilder AddNewtonsoftJsonFeatures(this IMvcCoreBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            AddServicesCore(builder.Services);
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures Newtonsoft.Json specific features such as input and output formatters.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcCoreBuilder"/>.</param>
+        /// <param name="setupAction">Callback to configure <see cref="MvcNewtonsoftJsonOptions"/>.</param>
+        /// <returns>The <see cref="IMvcCoreBuilder"/>.</returns>
+        public static IMvcCoreBuilder AddNewtonsoftJsonFeatures(
+            this IMvcCoreBuilder builder,
+            Action<MvcNewtonsoftJsonOptions> setupAction)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (setupAction == null)
+            {
+                throw new ArgumentNullException(nameof(setupAction));
+            }
+
+            AddServicesCore(builder.Services);
+
+            builder.Services.Configure(setupAction);
+
+            return builder;
+        }
+
+        // Internal for testing.
+        internal static void AddServicesCore(IServiceCollection services)
+        {
+            services.TryAddEnumerable(
+                ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, NewtonosftJsonMvcOptionsSetup>());
+            services.TryAddEnumerable(
+                ServiceDescriptor.Transient<IApiDescriptionProvider, JsonPatchOperationsArrayProvider>());
+            services.TryAddSingleton<IActionResultExecutor<JsonResult>, JsonResultExecutor>();
+
+            var tempDataSerializer = services.FirstOrDefault(
+                f => f.ServiceType == typeof(TempDataSerializer) && 
+                f.ImplementationType == typeof(DefaultTempDataSerializer));
+
+            if (tempDataSerializer != null)
+            {
+                // Replace the default implementation of TempDataSerializer
+                services.Remove(tempDataSerializer);
+            }
+            services.TryAddSingleton<TempDataSerializer, BsonTempDataSerializer>();
+
+            //
+            // JSON Helper
+            //
+            var jsonHelper = services.FirstOrDefault(
+                f => f.ServiceType == typeof(IJsonHelper) &&
+                f.ImplementationType == typeof(DefaultJsonHelper));
```

Comment:
A bit unfortunate, but I had to make `DefaultJsonHelper` and `DefaultTempDataSerialzers` public so that they could be referenced here. I wanted to avoid giving IVT access from Mvc to the new assembly

Reply:
You could use type full names.

Reply:
I think that's a better tradeoff